Assets/Scripts/Weapons/LookAtMouse.cs

```diff
@@ -11,8 +11,11 @@ public class LookAtMouse : MonoBehaviour
 
     private void Update()
     {
-        Vector3 mousePosition = Camera.ScreenToWorldPoint(Input.mousePosition);
-        Vector3 relativePoint = transform.position - mousePosition;
-        transform.rotation = Quaternion.Euler(0, 0, Mathf.Atan2(relativePoint.y, relativePoint.x) * Mathf.Rad2Deg + 90);
+        if(!PauseMenu.GameIsPaused)
+        {
+            Vector3 mousePosition = Camera.ScreenToWorldPoint(Input.mousePosition);
+            Vector3 relativePoint = transform.position - mousePosition;
+            transform.rotation = Quaternion.Euler(0, 0, Mathf.Atan2(relativePoint.y, relativePoint.x) * Mathf.Rad2Deg + 90);
+        }

```

Comment:
@Galaxy25 Is this the best way of doing this? Essentially we are putting a `isPaused` check at the beginning of each script update function. Does Unity not have a built in way of doing this?

Reply:
@Galaxy25 Not each script. Most Unity scripts are based off time. Say the enemy is firing a bullet time is what telling the enemy to shoot again. The only thing which doesn't work with time is player inputs.